src/main/java/frc/robot/subsystems/drivetrain/SwerveModule.java

```diff
@@ -0,0 +1,183 @@
+package frc.robot.subsystems.drivetrain;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.FeedbackDevice;
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.SupplyCurrentLimitConfiguration;
+import com.ctre.phoenix.motorcontrol.can.TalonFX;
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Ports;
+import frc.robot.UnitModel;
+import frc.robot.Utils;
+
+public class SwerveModule extends SubsystemBase {
+    private SupplyCurrentLimitConfiguration currLimitConfig = new SupplyCurrentLimitConfiguration(
+            Constants.ENABLE_CURRENT_LIMIT,
+            Constants.SwerveDrive.MAX_CURRENT,
+            Constants.SwerveModule.TRIGGER_THRESHOLD_CURRENT,
+            Constants.SwerveModule.TRIGGER_THRESHOLD_TIME
+    );
+    private final TalonFX driveMotor;
+    private final TalonSRX angleMotor;
+    private final int wheel;
+
+    private UnitModel unitDrive = new UnitModel(Constants.SwerveDrive.TICKS_PER_METER);
+    private UnitModel unitAngle = new UnitModel(Constants.SwerveDrive.TICKS_PER_RAD);
+
+    public SwerveModule(int wheel, TalonFX driveMotor, TalonSRX angleMotor, boolean[] inverted) {
+        // configure feedback sensors
+        angleMotor.configSelectedFeedbackSensor(FeedbackDevice.Analog, 0, Constants.TALON_TIMEOUT);
+        angleMotor.configFeedbackNotContinuous(Ports.SwerveDrive.IS_NOT_CONTINUOUS_FEEDBACK, Constants.TALON_TIMEOUT);
+
+        angleMotor.setNeutralMode(NeutralMode.Brake);
+
+        // set inversions
+        angleMotor.setInverted(inverted[0]);
+        driveMotor.setInverted(inverted[1]);
+
+        angleMotor.setSensorPhase(inverted[2]);
+        driveMotor.setSensorPhase(inverted[3]);
+
+        // Set amperage limits
+        driveMotor.configSupplyCurrentLimit(currLimitConfig);
+
+        angleMotor.configContinuousCurrentLimit(Constants.SwerveDrive.MAX_CURRENT);
+        angleMotor.enableCurrentLimit(Constants.ENABLE_CURRENT_LIMIT);
+
+        configPIDF();
+
+        // set voltage compensation and saturation
+        driveMotor.enableVoltageCompensation(Constants.ENABLE_VOLTAGE_COMPENSATION);
+        driveMotor.configVoltageCompSaturation(Constants.VOLTAGE_SATURATION);
+
+        angleMotor.enableVoltageCompensation(Constants.ENABLE_VOLTAGE_COMPENSATION);
+        angleMotor.configVoltageCompSaturation(Constants.VOLTAGE_SATURATION);
+
+        angleMotor.selectProfileSlot(0, 0);
+        driveMotor.selectProfileSlot(1, 0);
```

Comment:
There was a bug related to this, it works and I don't want to change it

Reply:
That sounds pretty weird. What was the bug?
I recommend checking again because now you need to use that `1` everywhere. If there really is a bug, report it to CTRE and benefit the whole FRC community 🙂.